src/vecdb/vdb_thread.rs

```diff
@@ -216,7 +216,7 @@ async fn vectorize_thread(
     constants: VecdbConstants,
     api_key: String,
     tokenizer: Arc<StdRwLock<Tokenizer>>,
-    gcx_weak: Weak<ARwLock<GlobalContext>>,
+    gcx: Arc<ARwLock<GlobalContext>>,
```

Comment:
why would you change that
this changes how long gcx lives

Reply:
I need gcx after that in order to pass it to the method to read files, since it contains the settings, an upgrade later could be used but I don't think it is a good idea because gcx could not be there anymore.

Reply:
you can always upgrade it back